AirportDepartures.playground/Contents.swift

```diff
@@ -16,9 +16,62 @@ import UIKit
 //: e. Use a `String?` for the Terminal, since it may not be set yet (i.e.: waiting to arrive on time)
 //:
 //: f. Use a class to represent a `DepartureBoard` with a list of departure flights, and the current airport
-
-
-
+enum FlightStatus: String {
+    case EnRoute = "En Route"
+    case Scheduled = "Scheduled"
+    case Canceled = "Canceled"
+    case Delayed = "Delayed"
+    case Boarding = "Boarding"
+}
+
+struct Airport {
+    let name: String
+}
+
+struct Flight {
+    var status: FlightStatus
+    var terminal: String? //Terminal "waiting to arrive on time" or #
+    var time: Date?
+    var destination: String
+    var airline: String
+    var flightID: String
+    
+}
+
+class DepartureBoard {
+    let currentAirport: Airport
+    var departureFlights: [Flight]
+    
+    init(currentAirport: Airport) {
+        self.currentAirport = currentAirport
+        self.departureFlights = []
+    }
+    
+    func upcomingFlightAlert(flight: Flight) {
+        switch flight.status {
+            case .Boarding:
+                if let flightTerminal = flight.terminal {
+                    print("Your flight is boarding, please head to terminal: \(flightTerminal) immediately. The doors are closing soon.")
+                } else {
+                    print("See the nearest information desk for more details.")
+                }
+            case .Canceled:
+                print("We're sorry your flight to (city) was canceled, here is a $500 voucher")
+            case .Delayed:
+                print("We're sorry, your flight is delayed.")
+            case .EnRoute:
+                print("Your flight is en route.")
+            case .Scheduled:
+                if let flightTime = flight.time {
+                    if let flightTerminal = flight.terminal {
+                        print("Your flight to \(flight.destination) is scheduled to depart at \(flightTime) from terminal: \(flightTerminal)")
+                    } else {
+                        print("See the nearest information desk for more details.")
+                    }
+                }
+        }
```

Comment:
Nice switch statement! I will say that convention in Swift is for the cases to be at the same indentation as the switch keyword. ctrl-i in Xcode produces this:

``` swift
func upcomingFlightAlert(flight: Flight) {
    switch flight.status {
    case .Boarding:
        if let flightTerminal = flight.terminal {
            print("Your flight is boarding, please head to terminal: \(flightTerminal) immediately. The doors are closing soon.")
        } else {
            print("See the nearest information desk for more details.")
        }
    case .Canceled:
        print("We're sorry your flight to (city) was canceled, here is a $500 voucher")
    case .Delayed:
        print("We're sorry, your flight is delayed.")
    case .EnRoute:
        print("Your flight is en route.")
    case .Scheduled:
        if let flightTime = flight.time {
            if let flightTerminal = flight.terminal {
                print("Your flight to \(flight.destination) is scheduled to depart at \(flightTime) from terminal: \(flightTerminal)")
            } else {
                print("See the nearest information desk for more details.")
            }
        }
    }
}
```